GitCommands/Git/GitCommandHelpers.cs

```diff
@@ -677,10 +677,11 @@ public static string CleanUpCmd(bool dryrun, bool directories, bool nonignored,
             return args.ToString();
         }
 
-        public static string GetAllChangedFilesCmd(bool excludeIgnoredFiles, UntrackedFilesMode untrackedFiles, IgnoreSubmodulesMode ignoreSubmodules = IgnoreSubmodulesMode.None)
+        public static string GetAllChangedFilesCmd(bool excludeIgnoredFiles, UntrackedFilesMode untrackedFiles, IgnoreSubmodulesMode ignoreSubmodules = IgnoreSubmodulesMode.None, bool noLocks = false)
         {
             var args = new ArgumentBuilder
             {
+                { noLocks && VersionInUse.SupportNoOptionalLocks, "--no-optional-locks" },
```

Comment:
I would remove `VersionInUse.SupportNoOptionalLock` from here and move it to:
`string command = GitCommandHelpers.GetAllChangedFilesCmd(..., VersionInUse.SupportNoOptionalLocks);`

Reply:
This is a guard to not include illegal options for a certain Git version, so it definetly belongs here.
Similar for other VersionInUse.

Reply:
From SR principle POV, this is a command builder and should be agnostic of git version. The onus is on the caller to supply correct parameters.
This way we can test the implementation.

Reply:
Then we need to add another layer above the command builder, to check that the arguments are OK for the Git version in use.
Changes need to be done to maybe 10 different places.

Reply:
> Why? The parameter is driven by the "is supported" flag....

I do not trust the client/application (in this case GitStatusMonitor) to only request valid options. This is extremely hard to review and test. (Is anyone explicitly testing older Git versions?) The server/command handler must check the arguments. This can be done outside the command builder, but still need to be done in GetAllChangedFilesCmd(). The argument builder can be moved to a separate function if that simplifies testing.